src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -20,6 +23,344 @@
 from mozanalysis.utils import get_time_intervals
 
 
+class ResultsHolder(UserDict):
+    """
+    Object to hold results from different methods.  It extends
+    the dictionary objects so that users can interact with it
+    like a dictionary with the same keys/values as before, making
+    it backward compatible.
+    """
+
+    def __init__(
+        self, *args, metrics: dict | None = None, params: dict | None = None, **kwargs
+    ):
+        """
+        Args:
+            metrics (list, optional): List of metrics used to generate the results.
+                                     Defaults to None.
+            params (dict, optional): Parameters used to generated the results.
+                                    Defaults to None.
+        """
+        super().__init__(*args, **kwargs)
+        self._metrics = metrics
+        self._params = params
+        # create this attribute to hold only the results data
+        # this dict is what was returned from the sample size
+        # methods historically
+        self.data = {
+            k: v for k, v in self.data.items() if k not in ["metrics", "params"]
+        }
+
+    @property
+    def metrics(self):
+        """the metrics property"""
+        return self._metrics
+
+    @property
+    def params(self):
+        """the self property"""
```

Comment:
```suggestion
        """the params property"""
```

Reply:
or maybe copy the descriptions from the __init__ docstrings into these ones?

Reply:
This is a really funny mistake, I like the idea of using the descriptions from the init docstring